cassovary-core/src/main/scala/com/twitter/cassovary/util/io/ListOfEdgesGraphReader.scala

```diff
@@ -114,25 +114,32 @@ class ListOfEdgesGraphReader[T](
 
     lazy val edgesIterator = edgesBySource.entrySet().iterator()
 
-    override def hasNext: Boolean = edgesIterator.hasNext
-
-    override def next(): NodeIdEdgesMaxId = {
-      try {
-        val elem = edgesIterator.next()
-        holder.id = elem.getKey
-        holder.edges = elem.getValue.toArray
-        holder.maxId = nodeMaxOutEdgeId.get(elem.getKey)
-        holder
-      } catch {
-        case NonFatal(exc) =>
-          throw new IOException("Parsing failed near line: %d in %s"
-            .format(lastLineParsed, filename), exc)
+    override def iterator: Iterator[NodeIdEdgesMaxId] = new Iterator[NodeIdEdgesMaxId] {
+
+      override def hasNext: Boolean = edgesIterator.hasNext
+
+      override def next(): NodeIdEdgesMaxId = {
+        try {
+          val elem = edgesIterator.next()
+          holder.id = elem.getKey
+          holder.edges = elem.getValue.toArray
+          holder.maxId = nodeMaxOutEdgeId.get(elem.getKey)
+          holder
+        } catch {
+          case NonFatal(exc) =>
+            throw new IOException("Parsing failed near line: %d in %s"
+              .format(lastLineParsed, filename), exc)
+        }
       }
     }
+
+    override def close(): Unit = {
+      Source.fromFile(filename).close()
+    }
   }
 
   def oneShardReader(filename: String): Iterator[NodeIdEdgesMaxId] = {
```

Comment:
correct me @szymonm if I am wrong, but we want to change the return value here to NodeStream. Then call NodeStream.close() after all the lines have been read during the array construction. 

And yes @AnishShah pls remove the () => everywhere after this change.